### DISCUSSION TITLE
load autocrop.lua only if profile condition is met

### DISCUSSION THREAD

**Discussion by geextahslex** [mpv-player/mpv#14463] [Q&A]
Hi, I would like to load autocrop.lua only when the profile condition (see below) is met. But all scripts load automatically on startup. How can I achieve the desired effect?

Thank you :)

```
[OSC2]
profile-cond=width / height >=1.77 and width / height <2 and get("video-params/primaries") == "bt.2020" 
```

**Comment by llyyr** [comment]
https://github.com/mpv-player/mpv/blob/master/TOOLS/lua/autocrop.lua#L29

Set `script-opts-append=autocrop-auto=no` in your config, then `script-opts-append=autocrop-auto=yes` in that profile

**Reply by geextahslex** [reply]
So I put `script-opts-append=autocrop-auto=no` in the dafult mpv.config and `script-opts-append=autocrop-auto=yes` in the profile. It runs still on all videos
I check the profile with `hue=50` to be sure that the profile is kicking in.

**Reply by Samillion** [reply]
> So I put `script-opts-append=autocrop-auto=no` in the dafult mpv.config and `script-opts-append=autocrop-auto=yes` in the profile. It runs still on all videos I check the profile with `hue=50` to be sure that the profile is kicking in.

Try adding `profile-restore=copy`

**Reply by kasper93** [reply]
scripts-opts doesn't work well in profiles. Scripts run in parallel and autoprofile is also a script, so there is no way of ensuring that the value will update before the autoload.lua is used.

**Reply by guidocella** [reply]
`autocrop.lua` doesn't enable runtime updates at all.

**Reply by geextahslex** [reply]
I ended up binding autocrop to a hotkey and use it manually if needed
